src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -1144,6 +1154,14 @@ bool AdaptDispatch::ChangeAttributesRectangularArea(const VTInt top, const VTInt
     changeOps.foreground = foregroundChanged ? std::optional{ foreground } : std::nullopt;
     changeOps.background = backgroundChanged ? std::optional{ background } : std::nullopt;
 
+    const auto underlineColor = allAttrsOff.GetUnderlineColor();
+    const auto underlineColorChanged = !underlineColor.IsDefault() || allAttrsOn.GetUnderlineColor().IsDefault();
+    changeOps.underlineColor = underlineColorChanged ? std::optional{ underlineColor } : std::nullopt;
+
+    const auto underlineStyle = allAttrsOff.GetUnderlineStyle();
+    const auto underlineStyleChanged = allAttrsOff.IsUnderlined() || !allAttrsOn.IsUnderlined();
+    changeOps.underlineStyle = underlineStyleChanged ? std::optional{ underlineStyle } : std::nullopt;
+
```

Comment:
I may need @j4james to cross-check this; I _think_ I understand what's going on!

Reply:
Sorry, I forgot to reply to this. But I think this bit should become unnecessary if we can get the underline style incorporated into the existing `CharacterAttributes` enum.